controlpanel/frontend/static/components/selectable-rows/selectable-rows.js

```diff
@@ -1,24 +1,92 @@
+/**
+ * Module to allow selecting entire rows in a table instead of using a small checkbox.
+ *
+ * It will disable any in-row actions buttons once a selection is made, but will re-enable them once
+ * all the selections are cleared.
+ *
+ * It can optionally support enabling a (global)
+ * button(s) that lives outside the table.
+ * How to use:
+ *  1. Add the class "selectable-rows" to your <table>
+ *  2. For any button you want to enable when there are selections add the class "selectable-rows__enable-on-selections"
+ *  3. Use the macro `row_selector` to put insert an invisible input box in the first column of your table
+ * The module will automatically disable any action buttons that are in side the row.
+ *
+ * WARNING: This module doesn't support more than 1 table on the same page
+ * @type {{init(): void, onSelectionChanged(*): void, baseClass: string, bindEvents(): void}}
+ */
+
 moj.Modules.selectable_rows = {
-  selector: '.selectable-rows tr',
+  baseClass: "selectable-rows",
 
   init() {
-    if (document.querySelector(this.selector)) {
+    this.tableRowSelector = `.${this.baseClass} tr`;
+    this.tableRowButtonSelector = `${this.tableRowSelector} button`;
+    this.selectionClass = `${this.baseClass}__row-selected`;
+    this.enableOnSelectionSelector = `.${this.baseClass}__enable-on-selections`;
+    this.selectionChangedEventName = `${this.baseClass}:rowSelectionChanged`;
+    if (document.querySelector(this.tableRowSelector)) {
       this.bindEvents();
     }
   },
 
+  /**
+   * Runs when rows are selected/deselected
+   * @param event
+   */
+  onSelectionChanged(event) {
+    const rowButtons = document.querySelectorAll(this.tableRowButtonSelector);
+    const enableOnSelectionButtons = document.querySelectorAll(
+      this.enableOnSelectionSelector
+    );
+    if (event.detail.rowsSelected.length > 0) {
+      // some rows are selected
+      rowButtons.forEach(e => (e.disabled = true));
+      enableOnSelectionButtons.forEach(e => e.disabled = false);
+    } else {
+      rowButtons.forEach(e => (e.disabled = false));
+      enableOnSelectionButtons.forEach(e => e.disabled = true);
+    }
+  },
+
   bindEvents() {
-    document.querySelectorAll(this.selector).forEach(row => {
+    window.addEventListener(this.selectionChangedEventName, e =>
+      this.onSelectionChanged(e)
+    );
+    document.querySelectorAll(this.tableRowSelector).forEach(row => {
       row.addEventListener("click", event => {
-        const checkbox = row.querySelector("input[type='checkbox'].row-selector");
+        const checkbox = row.querySelector(
+          "input[type='checkbox'].row-selector"
+        );
+
+        // user clicks on the button, don't select the row
+        if (event.target.tagName === "BUTTON") {
+          return true;
+        }
         if (checkbox) {
+          // user clicks somewhere on the row that isn't the checkbox itself
+          if (event.target !== checkbox) {
```

Comment:
I haven't tried it but it sounds like a good idea.
Is there something similar to `for` for `checkbox`es? it may be more predictable than handling this manually.

I was thinking about possible ways this could break if the raw contained links but in the case of customers that shouldn't be the case.

(what I mean by "break" is something like a user clicks on a link that opens a tab, the selection changes without him noticing, or something like that. But probably not a problem)

Reply:
There's nothing like `<label for="">` for a table row, this has to be done manually.

Reply:
Also, there must never be a link that opens a tab, or a selection that changes without user input - both definitely against UXD rules.